cirq-core/cirq/experiments/cross_entropy_benchmarking_test.py

```diff
@@ -25,127 +25,134 @@
 
+_DEPRECATION_MESSAGE = 'Use cirq.experiments.xeb_fitting.XEBCharacterizationResult instead'
+
 
 @pytest.mark.usefixtures('closefigures')
 def test_cross_entropy_benchmarking():
-    # Check that the fidelities returned from a four-qubit XEB simulation are
-    # close to 1 (deviations from 1 is expected due to finite number of
-    # measurements).
-    simulator = cirq.Simulator()
-    qubits = cirq.GridQubit.square(2)
-
-    # Sanity check single-qubit-gate causes error
-    with pytest.raises(ValueError):
-        build_entangling_layers(qubits, cirq.Z**0.91)
-
-    # Build a sequence of CZ gates.
-    interleaved_ops = build_entangling_layers(qubits, cirq.CZ**0.91)
-
-    # Specify a set of single-qubit rotations. Pick prime numbers for the
-    # exponent to avoid evolving the system into a basis state.
-    single_qubit_rots = [
-        [cirq.X**0.37],
-        [cirq.Y**0.73, cirq.X**0.53],
-        [cirq.Z**0.61, cirq.X**0.43],
-        [cirq.Y**0.19],
-    ]
-
-    # Simulate XEB using the default single-qubit gate set without two-qubit
-    # gates, XEB using the specified single-qubit gate set without two-qubit
-    # gates, and XEB using the specified single-qubit gate set with two-qubit
-    # gate. Check that the fidelities are close to 1.0 in all cases. Also,
-    # check that a single XEB fidelity is returned if a single cycle number
-    # is specified.
-    results_0 = cross_entropy_benchmarking(
-        simulator, qubits, num_circuits=3, repetitions=1000, cycles=range(4, 20, 5)
-    )
-    results_1 = cross_entropy_benchmarking(
-        simulator,
-        qubits,
-        num_circuits=3,
-        repetitions=1000,
-        cycles=[4, 8, 12],
-        scrambling_gates_per_cycle=single_qubit_rots,
-    )
-    results_2 = cross_entropy_benchmarking(
-        simulator,
-        qubits,
-        benchmark_ops=interleaved_ops,
-        num_circuits=3,
-        repetitions=1000,
-        cycles=[4, 8, 12],
-        scrambling_gates_per_cycle=single_qubit_rots,
-    )
-    results_3 = cross_entropy_benchmarking(
-        simulator,
-        qubits,
-        benchmark_ops=interleaved_ops,
-        num_circuits=3,
-        repetitions=1000,
-        cycles=15,
-        scrambling_gates_per_cycle=single_qubit_rots,
-    )
-    fidelities_0 = [datum.xeb_fidelity for datum in results_0.data]
-    fidelities_1 = [datum.xeb_fidelity for datum in results_1.data]
-    fidelities_2 = [datum.xeb_fidelity for datum in results_2.data]
-    fidelities_3 = [datum.xeb_fidelity for datum in results_3.data]
-    assert np.isclose(np.mean(fidelities_0), 1.0, atol=0.1)
-    assert np.isclose(np.mean(fidelities_1), 1.0, atol=0.1)
-    assert np.isclose(np.mean(fidelities_2), 1.0, atol=0.1)
-    assert len(fidelities_3) == 1
-
-    # Sanity test that plot runs.
-    ax = plt.subplot()
-    results_1.plot(ax)
+    with cirq.testing.assert_deprecated(_DEPRECATION_MESSAGE, deadline='v0.16', count=10):
```

Comment:
Could we use a few more contexts with lower counts that are a bit more tightly coupled to the things that are deprecated ? Just want to make sure we aren't accidentally raising two deprecations on some fxn calls.